src/runtime-helpers/runtime-helpers.ts

```diff
@@ -16,23 +16,36 @@
 module ScriptLab {
     const CODE_DIALOG_CLOSED_BY_USER = 12006;
 
+    const cachedAuthTokens: {
+        [clientIdAndResource: string]: {
+            token: string;
+            expiry: number;
+        }
+    } = {};
+
     /** [PREVIEW] Gets an access token on behalf of the user for a particular service
      * @param resource: The resource provider (default: 'graph' = Microsoft Graph; but could also be custom URI)
     */
     export function getAccessToken(clientId: string, resource: string = 'graph'): Promise<string> {
-        if (_isPlainWeb()) {
-            return _getAccessTokenViaWindowOpen(clientId, resource);
-        } else if (_isDialogApiSupported()) {
-            return _getAccessTokenViaDialogApi(clientId, resource);
+        const cachedAccessToken = _getCachedAccessToken(clientId, resource);
+        if (cachedAccessToken) {
+            return Promise.resolve(cachedAccessToken);
         } else {
```

Comment:
You don't need an "else" if your code does a "return" in the if statement.  It would never have reached the "else" statement stuff anyway, if it already returned out earlier.
So you can remove the "else" and un-indent that code block.